docs/openapi.yaml

```diff
@@ -0,0 +1,1143 @@
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+        - in: query
+          name: startTime
+          schema:
+            type: string
+            format: date-time
+            description: Minimum start time for scheduled missions
+          required: false
+        - in: query
+          name: endTime
+          schema:
+            type: string
+            format: date-time
+            description: Maximum end time for scheduled missions
+          required: false
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Schedule"
+          description: Request successful
+        "404":
+          description: Not found - No robot with id exists
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/event:
+    post:
+      operationId: postRobotEvent
+      summary: Create new event for robot
+      description: |
+        ### Overview
+        Adds a new event to the robot schedule
+
+        New entries to the schedule can be added as long as they don't conflict with already scheduled events.
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+      requestBody:
+        description: Time entry update
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/EventResponse"
+            examples:
+              scheduleEvent:
+                $ref: "#/components/examples/scheduleEvent"
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/EventResponse"
+          description: Request successful
+        "403":
+          description: Forbidden - Client not authorized to add events
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/event/{event_id}:
+    delete:
+      operationId: deleteRobotEvent
+      summary: Remove an event from the robot schedule
+      description: |
+        ### Overview
+        Deletes an event from the robot schedule.
```

Comment:
Does this cancel ongoing missions? Should we add another endpoint to abort mission/clear active mission?

Reply:
Saw the stop mission further down